profiling/src/api.rs

```diff
@@ -10,6 +10,11 @@ pub struct ValueType<'a> {
     pub r#type: &'a str,
     pub unit: &'a str,
 }
+impl<'a> ValueType<'a> {
+    pub fn new(r#type: &'a str, unit: &'a str) -> Self {
+        Self { r#type, unit }
+    }
+}
```

Comment:
Really minor: Since this is a trivial constructor, is it useful to keep? We don't seem to have constructors for most of the other structs below, and even in this PR there's a mix of constructing the struct via `new` and directly.

Reply:
I think so, because it fits on 1 line in many cases, but directly constructing it will take 4 lines. That's honestly why I made it. Thoughts?

Reply:
Sounds reasonable. In that case, maybe worth a quick pass on the PR to make sure we're using it everywhere we can perhaps?